src/SFA.DAS.ApplyService.Web/Controllers/Roatp/RoatpWhosInControlApplicationController.cs

```diff
@@ -74,8 +84,59 @@ public async Task<IActionResult> StartPage(Guid applicationId)
             return await AddPeopleInControl(applicationId);
         }
 
+        [HttpGet("refresh-directors-pscs")]
+        public async Task<IActionResult> RefreshDirectorsPscs(Guid applicationId, string ukprn, string companyNumber)
+        {
+            try
+            {
+                var companyDetails = await _companiesHouseApiClient.GetCompanyDetails(companyNumber);
+
+                if (companyDetails != null)
+                {
+                    switch (companyDetails.Status)
+                    {
+                        case CompaniesHouseSummary.ServiceUnavailable:
+                            _logger.LogInformation($"Issue refreshing directors/pscs - applicationId {applicationId} | Company Number : {companyNumber} | Status : Service Unavailable");
+                            return RedirectToAction("CompaniesHouseNotAvailable", "RoatpShutterPages");
+                        case CompaniesHouseSummary.CompanyStatusNotFound:
+                            _logger.LogInformation(
+                                $"Issue refreshing directors/pscs - applicationId {applicationId} | Company Number : {companyNumber} | Status : Company Status Not Found");
+                            return RedirectToAction("CompanyNotFound", "RoatpShutterPages");
+                    }
+
+                    if (!CompaniesHouseValidator.CompaniesHouseStatusValid(companyDetails.CompanyNumber, companyDetails.Status))
+                    {
+                        _logger.LogInformation($"Issue refreshing directors/pscs - applicationId {applicationId} | Company Number : {companyDetails.CompanyNumber} | Status : Companies House status not valid: {companyDetails.Status}");
+                        return RedirectToAction("CompanyNotFound", "RoatpShutterPages");
+                    }
+
+                    var applicationDetails = new Domain.Roatp.ApplicationDetails { CompanySummary = companyDetails };
+
+                    var organisation = await _applicationApiClient.GetOrganisationByUkprn(ukprn);
```

Comment:
Comment - Does this api call handle null ok?

Reply:
discussed that this pattern was lifted from elsewhere so better to keep with the existing behaviour for now...